src/compiler/cpp_generator.cc

```diff
@@ -993,7 +995,23 @@ void PrintHeaderServerMethodCallback(
         "controller) {\n"
         "               return this->$"
         "Method$(context, request, response, controller);\n"
-        "             }));\n");
+        "             }, nullptr));\n}\n");
+    printer->Print(
+        *vars,
+        "void SetMessageAllocatorFor_$Method$(\n"
+        "    ::grpc::MessageAllocator<$RealRequest$, $RealResponse$>* "
+        "allocator) {\n"
+        "  ::grpc::Service::experimental().MarkMethodCallback($Idx$,\n"
+        "    new ::grpc::internal::CallbackUnaryHandler< "
+        "$RealRequest$, $RealResponse$>(\n"
+        "      [this](::grpc::ServerContext* context,\n"
+        "             const $RealRequest$* request,\n"
+        "             $RealResponse$* response,\n"
+        "             ::grpc::experimental::ServerCallbackRpcController* "
+        "controller) {\n"
+        "               return this->$"
+        "Method$(context, request, response, controller);\n"
+        "             }, allocator));\n");
```

Comment:
There will be another variant of this in the near future, but it's not ready yet. It is possible that we will be moving server callback unary to a reactor model either on an optional or mandatory basis. As a result, any chance of  implementing this using a new setter in CallbackUnaryHandler rather than overriding the CallbackUnaryHandler altogether like this?